internal/auditd/auditd.go

```diff
@@ -80,6 +75,8 @@ func (o *Auditd) Read(ctx context.Context) error {
 
 	tracker := newSessionTracker(o.EventW)
 
+	defer closeReassembler(reassembler, reassembleAuditdEvents, tracker)
```

Comment:
It might not be a bad idea to add a unit test that attempts to simulate the failure we encountered. What do you think?

Reply:
Sure, I can do that.